test/TestProviderRound.spec.js

```diff
@@ -47,34 +63,29 @@ contract('ProviderRound', accounts => {
       );
     });
 
-    it('should set totalBondedAmount to 0', async () => {
-      const firstProvider = await providerRound.providers.call(accounts[0]);
-      assert.equal(0, firstProvider[5]); // [5] is totalBondedAmount
-    });
-
     it('should fail if not called during an active round', async () => {
       await blockMiner.mineUntilEndOfElectionPeriod(providerRound);
-      await assertFail( providerRound.provider(22, 10, 1, 25) );
+      await assertFail( providerRound.provider(22, 10, 1, 25, {from: accounts[0]}) );
     });
 
-    it('should register parameters before the lock period of an active round', async () => {
+    it('should work if called before the lock period of an active round', async () => {
       await blockMiner.mineUntilLastBlockBeforeLockPeriod(providerRound);
-      await providerRound.provider(22, 10, 1, 25);
+      await providerRound.provider(22, 10, 1, 25, {from: accounts[0]});
```

Comment:
We should have an assertion here to validate that the provider has been registered.